File: src/main/java/de/danielmaile/aether/config/ConfigManager.java
package de.danielmaile.aether.config;

import de.danielmaile.aether.Aether;
import org.bukkit.configuration.file.YamlConfiguration;

import java.io.File;

public class ConfigManager
{
    public LanguageManager getLanguageManager()
    {
        return languageManager;
    }

    public float getDungeonEndPartChance()
    {
        return dungeonEndPartChance;
    }

    public int getDungeonPartCap()
    {
        return dungeonPartCap;
    }

    public float getDungeonProbability()
    {
        return dungeonProbability;
    }

    public float getTreeProbability()
    {
        return treeProbability;
    }

    public float getLlamaJumpHeight()
    {
        return llamaJumpHeight;
    }

    public float getLlamaSpeed()
    {
        return llamaSpeed;
    }

    private LanguageManager languageManager;

    public boolean isItemConverterEnabled()
    {
        return itemConverterEnabled;
    }

    private boolean itemConverterEnabled;
    private float dungeonEndPartChance;
    private int dungeonPartCap;
    private float dungeonProbability;
    private float treeProbability;
    private float llamaJumpHeight;
    private float llamaSpeed;

    public ConfigManager()
    {
        load();
    }

    public void load()
    {
        File file = new File(Aether.getInstance().getDataFolder().getAbsolutePath() + File.separator
                + "locales" + File.separator + Aether.getInstance().getConfig().getString("language_file") + ".yml");
        if (!file.exists())
        {
            Aether.logError("Language file " + Aether.getInstance().getConfig().getString("language_file") + ".yml was not found!");
            file = new File(Aether.getInstance().getDataFolder().getAbsolutePath() + File.separator
                    + "locales" + File.separator + "de.yml");
        }

        languageManager = new LanguageManager(YamlConfiguration.loadConfiguration(file));
        itemConverterEnabled = Aether.getInstance().getConfig().getBoolean("item_converter");
        dungeonEndPartChance = (float) Aether.getInstance().getConfig().getDouble("world_generation.dungeons.end_part_chance");
        dungeonProbability = (float) Aether.getInstance().getConfig().getDouble("world_generation.dungeons.probability");
        dungeonPartCap = Aether.getInstance().getConfig().getInt("world_generation.dungeons.part_cap");
        treeProbability = (float) Aether.getInstance().getConfig().getDouble("world_generation.trees.probability");
        llamaJumpHeight = (float) Aether.getInstance().getConfig().getDouble("entities.llama.jump_height");
        llamaSpeed = (float) Aether.getInstance().getConfig().getDouble("entities.llama.speed");
    }
}

File: src/main/java/de/danielmaile/mpp/config/ConfigManager.kt
package de.danielmaile.mpp.config

import de.danielmaile.mpp.inst
import de.danielmaile.mpp.util.getResource
import de.danielmaile.mpp.util.logError
import org.bukkit.configuration.file.YamlConfiguration
import java.io.File
import java.io.InputStreamReader

class ConfigManager {

    val languageManager: LanguageManager
    val itemConverterEnabled: Boolean
    val dungeonEndPartChance: Float
    val dungeonPartCap: Int
    val dungeonProbability: Float
    val treeProbability: Float
    val llamaJumpHeight: Double
    val llamaSpeed: Float
    val minDungeonChestChecks: Int
    val maxDungeonChestChecks: Int
    val resourcePackHostIP: String
    val resourcePackHostPort: Int

    init {
        var file = File(
            inst().dataFolder.absolutePath + File.separator
                    + "locales" + File.separator + inst().config.getString("language_file") + ".yml"
        )
        if (!file.exists()) {
            logError("Language file " + inst().config.getString("language_file") + ".yml was not found!")
            file = File(
                (inst().dataFolder.absolutePath + File.separator
                        + "locales" + File.separator + "en.yml")
            )
        }

        val languageFile = YamlConfiguration.loadConfiguration(file)

        // Use locale files included in the plugin as fallback file
        // If the plugin doesn't include the language file provided in the config use the english file
        val fallBackLanguageFileStream = getResource("locales/${inst().config.getString("language_file")}.yml") ?: getResource("locales/en.yml")!!
        val fallBackLanguageFile = YamlConfiguration.loadConfiguration(InputStreamReader(fallBackLanguageFileStream, Charsets.UTF_8))
        languageFile.setDefaults(fallBackLanguageFile)
        languageManager = LanguageManager(languageFile)

        itemConverterEnabled = inst().config.getBoolean("item_converter")
        dungeonEndPartChance = inst().config.getDouble("world_generation.dungeons.end_part_chance").toFloat()
        dungeonProbability = inst().config.getDouble("world_generation.dungeons.probability").toFloat()
        dungeonPartCap = inst().config.getInt("world_generation.dungeons.part_cap")
        treeProbability = inst().config.getDouble("world_generation.trees.probability").toFloat()
        llamaJumpHeight = inst().config.getDouble("entities.llama.jump_height")
        llamaSpeed = inst().config.getDouble("entities.llama.speed").toFloat()
        minDungeonChestChecks = inst().config.getInt("world_generation.dungeons.chests.min_checks")
        maxDungeonChestChecks = inst().config.getInt("world_generation.dungeons.chests.max_checks")

        resourcePackHostIP = inst().config.getString("resource_pack.host.ip")!!
        resourcePackHostPort = inst().config.getInt("resource_pack.host.port")
    }
}